File: app/src/main/java/com/example/uxlab_animedxd/ui/home/BestMangaAdapter.java
package com.example.uxlab_animedxd.ui.home;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.example.uxlab_animedxd.R;
import java.util.List;

public class BestMangaAdapter extends RecyclerView.Adapter<BestMangaAdapter.ViewHolder> {

    private final List<MangaItem> mangaList;

    public BestMangaAdapter(List<MangaItem> mangaList) {
        this.mangaList = mangaList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_best_manga, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        MangaItem item = mangaList.get(position);
        holder.poster.setImageResource(item.getImageRes());
        holder.title.setText(item.getTitle());
        holder.subtitle.setText(item.getSubtitle());
    }

    @Override
    public int getItemCount() {
        return mangaList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        ImageView poster;
        TextView title;
        TextView subtitle;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            poster = itemView.findViewById(R.id.imgPoster);
            title = itemView.findViewById(R.id.tvTitle);
            subtitle = itemView.findViewById(R.id.tvSubtitle);
        }
    }
}

File: app/src/main/java/com/example/uxlab_animedxd/weebzone/adapter/BestMangaAdapter.kt
package com.example.uxlab_animedxd.weebzone.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.uxlab_animedxd.R
import com.example.uxlab_animedxd.weebzone.model.Manga

class BestMangaAdapter(private val data: List<Manga>) :
    RecyclerView.Adapter<BestMangaAdapter.VH>() {

    inner class VH(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val imgPoster: ImageView = itemView.findViewById(R.id.imgPoster)
        private val tvTitle:  TextView   = itemView.findViewById(R.id.tvTitle)
        private val tvSub:    TextView   = itemView.findViewById(R.id.tvSubtitle)

        fun bind(item: Manga) {
            imgPoster.setImageResource(item.imageRes)
            tvTitle.text = item.title.uppercase()
            tvSub.text   = item.description
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
        VH(
            LayoutInflater.from(parent.context)
                .inflate(R.layout.item_best_manga, parent, false)
        )

    override fun getItemCount() = data.size

    override fun onBindViewHolder(holder: VH, position: Int) =
        holder.bind(data[position])
}
